plugins/osc/README.developer.md

```diff
@@ -0,0 +1,675 @@
+OSC Plugin Developer Information
+================================
+
+Welcome to the OSC Plugin walkthrough. This is a step by step guide on how
+to write a plugin for OLA, using the OSC (Open Sound Control) plugin as an
+example. The OSC plugin uses [liblo](http://liblo.sourceforge.net/) to
+handle the OSC messages.
+
+This assumes you have a working knowledge of C++ including references and
+pointers as well as a basic understanding of the string, map and vector
+classes. Most of the documentation is inline with the code so please keep it
+up to date if you change anything.
+
+Before we begin you should do the following:
+
+* Read https://wiki.openlighting.org/index.php/OLA_developer_info which
+  provides an overview of some of the classes we'll use.
+* Have an understanding of the map & vector classes from the STL
+  (http://www.sgi.com/tech/stl/ or http://www.cplusplus.com/reference/stl/).
+* Know something about the
+  [OSC specification](http://opensoundcontrol.org/spec-1_0). You don't need
+  to know the wire format for the OSC messages. but you should at least
+  recognize OSC addresses and know the different data types in a message.
+* Look at the [liblo site](http://liblo.sourceforge.net/) and read the
+  examples.
+* Get familiar with the
+  [string class](http://www.cplusplus.com/reference/string/string/), since
+  we use that a bit.
+* Basic knowledge of shell scripting, which is used in the `./configure`
+  script.
+* Make sure you can complete a full OLA build & test cycle (`make`,
+  `make check`).
+
+The initial commit of this plugin was Rev
+`b9d6a47e9a2586d51dac521d7532fb11d76e0ae5`. It's worth having that diff
+open as you step through the explanations.
+
+Let's get to it. If you make it all the way through this doc and I ever meet
+you in person I'll buy you a beer!
```

Comment:
Heh, you need to talk to @nomis52 for that, and fly to the states! Or I'll buy you one if I see you in the UK.